.github/workflows/vm-kernel.yaml

```diff
@@ -21,22 +21,42 @@ jobs:
     outputs:
       image: ${{ fromJSON(steps.build-linux-kernel.outputs.metadata)['image.name'] }}@${{ steps.build-linux-kernel.outputs.digest }}
 
-    # Use the biggest GitHub Hosted runner
-    runs-on: gha-ubuntu-22.04-64cores
+    runs-on: [ self-hosted, gen3, large ]
     steps:
-
       - name: git checkout
         uses: actions/checkout@v3
 
+      - name: set custom docker config directory
+        run: |
+          mkdir -p .docker-custom
+          echo DOCKER_CONFIG=$(pwd)/.docker-custom >> $GITHUB_ENV
+      - name: login to docker hub
+        run: |
+          DOCKERHUB_AUTH=$(echo -n "${{ secrets.NEON_DOCKERHUB_USERNAME }}:${{ secrets.NEON_DOCKERHUB_PASSWORD }}" | base64)
+          echo "::add-mask::${DOCKERHUB_AUTH}"
+
+          cat <<-EOF > .docker-custom/config.json
+            {
+              "auths": {
+                "https://index.docker.io/v1/": {
+                  "auth": "${DOCKERHUB_AUTH}"
+                }
+              }
+            }
+          EOF
+
       - name: docker - install qemu
         uses: docker/setup-qemu-action@v2
       - name: docker - setup buildx
         uses: docker/setup-buildx-action@v2
-      - name: login to docker hub
-        uses: docker/login-action@v2
-        with:
-          username: ${{ secrets.NEON_DOCKERHUB_USERNAME }}
-          password: ${{ secrets.NEON_DOCKERHUB_PASSWORD }}
+
+      - name: get kernel version
+        id: get-kernel-version
+        run: |
+          linux_config=$(ls neonvm/hack/linux-config-*)  # returns something like "neonvm/hack/linux-config-6.1.63"
+          kernel_version=${linux_config##*-}             # returns something like "6.1.63"
```

Comment:
neat! Opened #663 to use this same technique for `make kernel` so that the filename can be an authoritative source for the kernel version we're using.

re: "maybe we should get it from another place" - yeah, I'm not totally sure either, but this can work for now :)